ext/crypto/00_crypto.js

```diff
@@ -4326,7 +4326,13 @@
             length,
           });
 
-          return buf.buffer;
+          // 8.
+          if (!length) {
+            return buf.buffer;
+          } else if (buf.buffer.byteLength * 8 < length || length % 8 !== 0) {
+            throw new DOMException("Invalid length", "OperationError");
+          }
+          return buf.buffer.slice(0, length / 8);
```

Comment:
In that case, don't use the casted for the null check. Otherwise you can throw when 0 is passed.

```suggestion
          // 8.
          if (arguments[2] === null) {
            return buf.buffer;
          }
          
          if (typeof length !== "number" || length === 0 || buf.buffer.byteLength * 8 < length || length % 8 !== 0) {
            throw new DOMException("Invalid length", "OperationError");
          }
          
          return buf.buffer.slice(0, length / 8);
```